session-01/validators.js

```diff
@@ -7,17 +7,64 @@
   - username cannot contain special characters
 */
 function validUsername(username) {
-  return;
-}
 
+  let unicodeNum = username.charCodeAt(0);
+
+    // checks if first character is unicode point value for upper or lower case character
+    if ((unicodeNum >= 65 && unicodeNum <= 90) || (unicodeNum >= 97 && unicodeNum <= 122)){
+        
+        //checks un length
+         if (username.length >= 3 && username.length <= 10){
+
+            //loops to find special character in whole string based on code point values
+            for(i = 1; i < username.length; i++){
+
+                if ( !((username.charCodeAt(i) >= 65 && username.charCodeAt(i) <= 90) || (username.charCodeAt(i) >= 97 && username.charCodeAt(i) <= 122) || (username.charCodeAt(i) >= 48 && username.charCodeAt(i) <= 57))) {
+                    return false;
+                }
+
+            }
+            return true;
+         }
+
+    } 
+
+  return false;
+}
```

Comment:
Consider using the guard clause pattern like in this example: https://github.com/CUNYTechPrep/js-summer-prep-2024/blob/main/session-01/validators.solution.js#L10-L16